src/main/java/graphql/validation/TraversalContext.java

```diff
@@ -190,29 +200,32 @@ private GraphQLArgument find(List<GraphQLArgument> arguments, String name) {
     @Override
     public void leave(Node node, List<Node> ancestors) {
         if (node instanceof OperationDefinition) {
-            outputTypeStack.remove(outputTypeStack.size() - 1);
+            pop(outputTypeStack);
         } else if (node instanceof SelectionSet) {
-            parentTypeStack.remove(parentTypeStack.size() - 1);
+            pop(parentTypeStack);
         } else if (node instanceof Field) {
             leaveName(((Field) node).getName());
-            fieldDefStack.remove(fieldDefStack.size() - 1);
-            outputTypeStack.remove(outputTypeStack.size() - 1);
+            pop(fieldDefStack);
+            pop(outputTypeStack);
         } else if (node instanceof Directive) {
             directive = null;
         } else if (node instanceof InlineFragment) {
-            outputTypeStack.remove(outputTypeStack.size() - 1);
+            pop(outputTypeStack);
         } else if (node instanceof FragmentDefinition) {
             leaveName(((FragmentDefinition) node).getName());
-            outputTypeStack.remove(outputTypeStack.size() - 1);
+            pop(outputTypeStack);
         } else if (node instanceof VariableDefinition) {
             inputTypeStack.remove(inputTypeStack.size() - 1);
```

Comment:
```suggestion
           pop(inputTypeStack);
```
I just assmed this was a refactoring miss.
Should it also pop the `defaultValueStack` as the other cases bellow?

Reply:
yes changed